### PR TITLE
generic ephemeral volume util, base code and controller

### PR DESCRIPTION
**Pull request by pohly**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Although small, sharing the code for pvc name creation and ownership check increases readability and consistency.

#### Special notes for your reviewer:

Let's first merge the utility package and convert one component (the kube-controller-manager) to use it. I'll submit PRs for the other areas later.

https://github.com/pohly/kubernetes/tree/generic-ephemeral-volume-util-full contains all commits.

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

### CODE REVIEW

**Comment by spiffxp** [comment]
/test pull-kubernetes-node-e2e-containerd
ref: #105381

File: pkg/volume/util/ephemeral/ephemeral.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+// indeed the PVC that was created for the Pod by calling IsUsable.
+func VolumeClaimName(pod *v1.Pod, volume *v1.Volume) string {
+	return pod.Name + "-" + volume.Name
+}
+
+// VolumeIsForPod checks that the PVC is the ephemeral volume that
+// was created for the Pod. It returns an error that is informative
+// enough to be returned by the caller without adding further details
+// about the Pod or PVC.
+func VolumeIsForPod(pod *v1.Pod, pvc *v1.PersistentVolumeClaim) error {
```
Comment (msau42) [inline pkg/volume/util/ephemeral/ephemeral.go:49]: Can you add some quick unit tests for this function?

File: pkg/volume/util/ephemeral/ephemeral.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
+/*
```
Comment (msau42) [inline pkg/volume/util/ephemeral/ephemeral.go:1]: It looks like there's a push to put any scheduler dependencies into component-helpers: https://github.com/kubernetes/kubernetes/pull/103062

Comment (pohly) [reply pkg/volume/util/ephemeral/ephemeral.go:1]: Then let's create `k8s.io/component-helpers/storage/ephemeral`.

File: pkg/volume/util/ephemeral/ephemeral.go (hunk 1 above)
Comment (pohly) [reply pkg/volume/util/ephemeral/ephemeral.go:49]: Will do.

**Comment by pohly** [comment]
@kubernetes/dep-approvers: please approve. Not a real dependency change, just a new helper package under staging.

**Pull request merged**